cloud/docker/docker.py

```diff
@@ -624,14 +624,14 @@ def __init__(self, module):
                 # host mount (e.g. /mnt:/tmp, bind mounts host's /tmp to /mnt in the container)
                 elif 2 <= len(parts) <= 3:
                     # default to read-write
-                    ro = False
+                    mode = 'rw'
                     # with supplied bind mode
                     if len(parts) == 3:
-                        if parts[2] not in ['ro', 'rw']:
-                            self.module.fail_json(msg='bind mode needs to either be "ro" or "rw"')
+                        if parts[2] not in ['ro', 'rw', 'z', 'Z']:
+                            self.module.fail_json(msg='bind mode needs to be "ro", "rw", "z" or "Z"')
                         else:
-                            ro = parts[2] == 'ro'
-                    self.binds[parts[0]] = {'bind': parts[1], 'ro': ro }
+                            mode = parts[2]
+                    self.binds[parts[0]] = {'bind': parts[1], 'mode': mode }
```

Comment:
What do you mean? Before my patch, ``ro`` was passed as a boolean between the part in line 613-636 and the part in line 1196-1205, but then passed as ``String mode`` to Docker on line 1205